plugins/notifiers/client.go

```diff
@@ -8,18 +8,30 @@ import (
 	"time"
 
 	"github.com/goto/guardian/pkg/log"
+	"github.com/goto/guardian/plugins/notifiers/lark"
+	"github.com/goto/guardian/plugins/notifiers/slack"
 	"github.com/mitchellh/mapstructure"
 
 	"github.com/goto/guardian/domain"
-	"github.com/goto/guardian/plugins/notifiers/slack"
 )
 
 type Client interface {
 	Notify(context.Context, []domain.Notification) []error
 }
 
+type NotifyManager struct {
+	clients []Client
+}
+
+func (m *NotifyManager) Notify(ctx context.Context, notification []domain.Notification) {
+	for _, client := range m.clients {
+		client.Notify(ctx, notification)
```

Comment:
we need to evaluate the criteria here before actually calling `client.Notify`. `m.clients` have to somehow still store the notifier config (criteria, custom messages) along with the client instance

Reply:
actually slack ws criteria already handled at with GetSlackWorkspaceForUser inside slack.Notify

Reply:
Now a client is for every notifier in config.Notifiers. And for slack clients we have the expression evaluated

Reply:
i don't think `criteria` expression is evaluated for lark right now? CMIIW